velox/vector/LazyVector.cpp

```diff
@@ -109,6 +109,9 @@ void LazyVector::ensureLoadedRows(
     SelectivityVector& baseRows) {
   decoded.decode(*vector, rows, false);
   if (decoded.base()->encoding() != VectorEncoding::Simple::LAZY) {
+    if (isLazyNotLoaded(*decoded.base())) {
```

Comment:
This will always load all rows however the objective of ensureLoadedRows() is to only load the rows that are passed in. This function also does not recognize nested lazy in a ROW so we will have to add special handling for that.

Reply:
@bikramSingh91 I will pass SelectivityVector to loadedVector() call, so it can handle that. Let me know if that won't work. 

> This function also does not recognize nested lazy in a ROW
Which function?

Reply:
loadedVector() as an API is not meant to load a subset of rows, its objective is to act as an accessor so it either returns an already loaded vector (which can be a partially loaded vector) or load the full set of rows if not loaded, (see comment above loadedVectorShared). Therefore, I would recommend reverting that part.

> 
> > This function also does not recognize nested lazy in a ROW
> 
> Which function?

I meant that the function "ensureLoadedLazy" does not recognize that ROW can have nested lazies, therefore it only handles a lazy or an encoded lazy (dictionary or constant encoded). Therefore, I was suggested something like this:

```
  if (decoded.base()->encoding() != VectorEncoding::Simple::LAZY) {
    if (decoded.base()->encoding() == VectorEncoding::Simple::ROW && isLazyNotLoaded(*decoded.base())) {
      auto children = decoded.base()->asUnchecked<RowVector>()->children();
      DecodedVector decodedChild;
      for (auto& child : children) {
        ensureLoadedRows(child, rows, decodedChild, baseRows);
      }
      decoded.base()->loadedVector();
    }
    return;
  }
```

This would ensure all children are loaded and the loadedVector() call to loadedVector() at the end will correctly init the internal state of the row vector, that is, replace the children with the loaded vectors.

Also, can you please add tests in LazyVectorTest that verify that only the selected subset of rows get loaded when ensureLoadedLazy() is called on the rowVector. Please see the test lazyInDoubleDictionary for reference.